scripts/globals/magic.lua

```diff
@@ -618,211 +467,124 @@ function applyResistanceEffect(player,spell,target,diff,skill,bonus,effect)
-    local sixteenth = ((1 - p)^4);
-    -- print("HALF:",half);
-    -- print("QUART:",quart);
-    -- print("EIGHTH:",eighth);
-    -- print("SIXTEENTH:",sixteenth);
-
-    local resvar = math.random();
-
-    -- Determine final resist based on which thresholds have been crossed.
-    if (resvar <= sixteenth) then
-        resist = 0.0625;
-        --printf("Spell resisted to 1/16!!!  Threshold = %u",sixteenth);
-    elseif (resvar <= eighth) then
-        resist = 0.125;
-        --printf("Spell resisted to 1/8!  Threshold = %u",eighth);
-    elseif (resvar <= quart) then
-        resist = 0.25;
-        --printf("Spell resisted to 1/4.  Threshold = %u",quart);
-    elseif (resvar <= half) then
-        resist = 0.5;
-        --printf("Spell resisted to 1/2.  Threshold = %u",half);
-    else
-        resist = 1.0;
-        --printf("1.0");
-    end
-
-    return resist;
-
+    return getMagicResist(p);
 end;
 
 --Applies resistance for additional effects
 function applyResistanceAddEffect(player,target,element,bonus)
 
-    local resist = 1.0;
-    local magicaccbonus = 0;
-
-    if (bonus ~= nil) then
-        magicaccbonus = magicaccbonus + bonus;
-    end
-
-    --get the base acc (just skill plus magic acc mod)
-    local magicacc = 0;
+    -- Add a base 75 magic acc
+    local p = getMagicHitRate(player, target, 0, element, 75, bonus);
 
-    --add acc for staves
-    local affinityBonus = AffinityBonus(player, element);
-    magicaccbonus = magicaccbonus + (affinityBonus-1) * 200;
+    return getMagicResist(p);
+end;
 
-    --base magic evasion (base magic evasion plus resistances(players), plus elemental defense(mobs)
-    local magiceva = target:getMod(resistMod[element]);
+function getMagicHitRate(caster, target, skillType, element, fullBonus, minorBonus)
+  -- resist everything if magic shield is active
+  if (target:hasStatusEffect(EFFECT_MAGIC_SHIELD, 0)) then
+    return 0;
+  end
+
+  local magicacc = 0;
+  local magiceva = 0;
+  local p = 0;
+
+  if(minorBonus == nil) then
+    minorBonus = 0;
+  end
+
+  -- Get the base acc (just skill + skill mod (79 + skillID = ModID) + magic acc mod)
+  local magicacc = caster:getMod(MOD_MACC);
+
+  if(skillType ~= 0) then
+    magicacc = magicacc + caster:getSkillLevel(skillType) + caster:getMod(79 + skillType);
+  end
+
+  magicacc = magicacc + fullBonus;
+
+  local resMod = 0; -- Some spells may possibly be non elemental, but have status effects.
+  if (element > ELE_NONE) then
+      resMod = target:getMod(resistMod[element]);
+  end
+
+  if (element > ELE_NONE) then
+    -- Add acc for staves
+    local affinityBonus = AffinityBonus(caster, element);
+    minorBonus = minorBonus + (affinityBonus-1) * 200;
+  end
+
+  --add acc for RDM group 1 merits
+  if (element > 0 and element <= 6) then
+    minorBonus = minorBonus + caster:getMerit(rdmMerit[element]);
+  end
+
+  -- BLU mag acc merits - nuke acc is handled in bluemagic.lua
+  if (skill == BLUE_SKILL) then

```

Comment:
Just checked, can't move it because blue spells use `applyResistanceEffect` for checking resistance and that doesn't pass through bluemagic.lua.

Anyway yeah, I don't want to touch too many things. I'll refactor more in smaller pieces.